lib/puppet/util/windows/sid.rb

```diff
@@ -24,14 +24,26 @@ def name_to_sid(name)
     # e.g. 'S-1-5-32-544'. The name can be specified as 'Administrators',
     # 'BUILTIN\Administrators', or 'S-1-5-32-544', and will return the
     # SID object. Returns nil if the account doesn't exist.
+    # This method returns a SID hash with the system, account, domain, SID, etc
     def name_to_sid_object(name)
       # Apparently, we accept a symbol..
       name = name.to_s.strip if name
 
+      # TODO: figure out who calls it and what they need
       # if it's in SID string form, convert to user
-      parsed_sid = Win32::Security::SID.string_to_sid(name) rescue nil
 
-      parsed_sid ? Win32::Security::SID.new(parsed_sid) : Win32::Security::SID.new(name)
+      # if name is a SID string, convert it to raw bytes for use with lookup_account_sid
+      raw_sid_bytes = nil
+      begin
+        string_to_sid_ptr(name) do |sid_ptr|
+          valid = ! sid_ptr.nil? && ! sid_ptr.null?
+          raw_sid_bytes = sid_ptr.read_array_of_uchar(get_length_sid(sid_ptr))
+        end
+      rescue
+      end
+
+      # TODO: using . for domain seems wrong if we accept DOMAIN\USER for name
+      raw_sid_bytes ? lookup_account_sid(nil, raw_sid_bytes) : lookup_account_name('.', name)
```

Comment:
Yeah, @glennsarti and I discussed this and decided we're going to file a new ticket for that particular problem, since we don't believe it has ever worked.